src/Adapters/Auth/ldap.js

```diff
@@ -12,23 +12,34 @@ function validateAuthData(authData, options) {
       );
     });
   }
+  const clientOptions = (options.url.startsWith("ldaps://")) ?
+    { url: options.url, tlsOptions: options.tlsOptions } : { url: options.url };
 
-  const client = ldapjs.createClient({ url: options.url });
+  const client = ldapjs.createClient(clientOptions);
   const userCn =
     typeof options.dn === 'string'
       ? options.dn.replace('{{id}}', authData.id)
       : `uid=${authData.id},${options.suffix}`;
 
   return new Promise((resolve, reject) => {
-    client.bind(userCn, authData.password, err => {
-      if (err) {
-        client.destroy(err);
-        return reject(
-          new Parse.Error(
-            Parse.Error.OBJECT_NOT_FOUND,
-            'LDAP: Wrong username or password'
-          )
-        );
+    let error;
```

Comment:
NIT: Could you please move this line to inside the `if` currently in line 27?

Reply:
can't move it in the `if` clause. the if clause is executed twice when the cert mismatch happens. first time the mismatch error is detected, second time the client destruction destroyes the correct error message. so the error has to be outside to be set to the first mismatch error after the second client destroy error happened.

Reply:
It does not make sense. The var is not used outside the scope and inside the scope it is only assigned and then used to reject the promise. You can also do the following just to make sure:

```js
reject(error);
client.destroy(ldapError);
return;
```

Reply:
You are right. The `return reject (error)` was the part that confuses the execution flow. With the rejection and the return splitted I could also remove the `case undefined`.